### DISCUSSION TITLE
How can I connect my implementation of persistent storage to Symfony Lock?

### DISCUSSION THREAD

**Discussion by voskobovich** [symfony/symfony#48115] [Q&A] [answered]
Hi all!
I need to store the lock state in Couchbase DB.

I implemented the `Symfony\Component\Lock\PersistingStoreInterface` as class `App\Lock\CouchbaseStore`. Now I need to connect it to my app.
I looked here `\Symfony\Bundle\FrameworkBundle\DependencyInjection\FrameworkExtension::registerLockConfiguration` and here` \Symfony\Bundle\FrameworkBundle\DependencyInjection\Configuration::addLockSection` but I didn't understand how to connect it to mine app. For example, can I redefine `\Symfony\Component\Lock\Store\StoreFactory` in a DI container or something else?

**Comment by MatTheCat** [comment, marked as answer]
Hi :wave: 

You can inject your `CouchbaseStore` in a `LockFactory` service and directly use it; no need to bother with the `StoreFactory` (unless you want to propose your implementation to Symfony!)

**Reply by voskobovich** [reply]
> Hi 👋
> 
> You can inject your `CouchbaseStore` in a `LockFactory` service and directly use it; no need to bother with the `StoreFactory` (unless you want to propose your implementation to Symfony!)

I thought about it. How will the env variable work with this solution? 

```
###> symfony/lock ###
# Choose one of the stores below
# postgresql+advisory://db_user:db_password@localhost/db_name
LOCK_DSN=flock
###< symfony/lock ###
```

Can I replace the `lock.factory` alias in DI container with my own?
Like this
```
$services->set('lock.factory', LockFactory::class)->args([CouchbaseStore::class]);
```

**Reply by MatTheCat** [reply]
You won’t need the env var in your case as you won’t use the FrameworkBundle’s integration. You can even disable it:

```yaml
framework:
    lock:
        enabled: false
        default: whatever # see https://github.com/symfony/symfony/pull/48119
```

Then indeed you can create the `LockFactory` service yourself:

```php
$services->set(LockFactory::class)->arg('$store', new Reference(CouchbaseStore::class));
```

**Reply by voskobovich** [reply]
Thanks for your help!
I didn``t know about `enabled: false`.
Now I solved my task.

**Reply by MatTheCat** [reply]
You’re welcome! Now you can mark [my comment](https://github.com/symfony/symfony/discussions/48115#discussioncomment-4062912) as an answer :ok_hand:
